packages-exp/performance-exp/index.ts

```diff
@@ -17,68 +17,32 @@
 
 import '@firebase/installations';
 
-import firebase from '@firebase/app';
-import { FirebaseApp, FirebaseNamespace } from '@firebase/app-types';
+import { FirebaseApp } from '@firebase/app-types';
 import { _FirebaseNamespace } from '@firebase/app-types/private';
-import { Component, ComponentType } from '@firebase/component';
-import { FirebaseInstallations } from '@firebase/installations-types';
-import { FirebasePerformance } from '@firebase/performance-types-exp';
+import {
+  FirebasePerformance,
+  PerformanceSettings
+} from '@firebase/performance-types-exp';
 
-import { name, version } from './package.json';
 import { PerformanceController } from './src/controllers/perf';
 import { setupApi } from './src/services/api_service';
 import { SettingsService } from './src/services/settings_service';
 import { ERROR_FACTORY, ErrorCode } from './src/utils/errors';
 
 const DEFAULT_ENTRY_NAME = '[DEFAULT]';
 
-export function registerPerformance(instance: FirebaseNamespace): void {
-  const factoryMethod = (
-    app: FirebaseApp,
-    installations: FirebaseInstallations
-  ): PerformanceController => {
-    if (app.name !== DEFAULT_ENTRY_NAME) {
-      throw ERROR_FACTORY.create(ErrorCode.FB_NOT_DEFAULT);
-    }
-    if (typeof window === 'undefined') {
-      throw ERROR_FACTORY.create(ErrorCode.NO_WINDOW);
-    }
-    setupApi(window);
-    SettingsService.getInstance().firebaseAppInstance = app;
-    SettingsService.getInstance().installationsService = installations;
-    return new PerformanceController(app);
-  };
-
-  // Register performance with firebase-app.
-  (instance as _FirebaseNamespace).INTERNAL.registerComponent(
-    new Component(
-      'performance',
-      container => {
-        /* Dependencies */
-        // getImmediate for FirebaseApp will always succeed
-        const app = container.getProvider('app').getImmediate();
-        // The following call will always succeed because perf has `import
-        // '@firebase/installations'`
-        const installations = container
-          .getProvider('installations')
-          .getImmediate();
-
-        return factoryMethod(app, installations);
-      },
-      ComponentType.PUBLIC
-    )
-  );
-
-  instance.registerVersion(name, version);
-}
-
-registerPerformance(firebase);
-
-declare module '@firebase/app-types' {
-  interface FirebaseNamespace {
-    performance?: { (app?: FirebaseApp): FirebasePerformance };
+export function getPerformance(
```

Comment:
Seems like we still need to register the component with the firebaseApp. Here is an example:
https://github.com/firebase/firebase-js-sdk/blob/master/packages-exp/functions-exp/src/config.ts

Reply:
We need to have installations available before we can do this, and we need to verify whether we need to actually register performance or not. Will leave out until we have more clarity.